### PR TITLE
Make jdbc-types-mapped-to-varchar case insensitive

### PR DESCRIPTION
**Pull request by kokosing**
Make jdbc-types-mapped-to-varchar case insensitive

### CODE REVIEW

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java [hunk 1]
```diff
@@ -301,7 +301,7 @@ protected static ResultSet getColumns(JdbcTableHandle tableHandle, DatabaseMetaD
 
     protected Optional<ColumnMapping> getForcedMappingToVarchar(JdbcTypeHandle typeHandle)
     {
-        if (typeHandle.getJdbcTypeName().isPresent() && jdbcTypesMappedToVarchar.contains(typeHandle.getJdbcTypeName().get())) {
+        if (typeHandle.getJdbcTypeName().isPresent() && jdbcTypesMappedToVarchar.contains(typeHandle.getJdbcTypeName().get().toLowerCase(ENGLISH))) {
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java]: `jdbcTypesMappedToVarchar` is protected, so it requires modications in subclasses.

instead, i'd consider using `TreeSet` with `java.lang.String#CASE_INSENSITIVE_ORDER` for `jdbcTypesMappedToVarchar`

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcConfig.java [hunk 2]
```diff
@@ -80,7 +83,9 @@ public BaseJdbcConfig setCaseInsensitiveNameMatchingCacheTtl(Duration caseInsens
     @Config("jdbc-types-mapped-to-varchar")
     public BaseJdbcConfig setJdbcTypesMappedToVarchar(String jdbcTypesMappedToVarchar)
     {
-        this.jdbcTypesMappedToVarchar = ImmutableSet.copyOf(Splitter.on(",").omitEmptyStrings().trimResults().split(nullToEmpty(jdbcTypesMappedToVarchar)));
+        this.jdbcTypesMappedToVarchar = Streams.stream(Splitter.on(",").omitEmptyStrings().trimResults().split(nullToEmpty(jdbcTypesMappedToVarchar)))
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcConfig.java]: nit: using `splitToList().stream()` would result in more natural code flow.

**Comment by kokosing** [comment]
@findepi Thank you for the review. I added two more very tiny commits here. Please take a look.

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java [hunk 3]
```diff
@@ -309,7 +309,9 @@ protected static ResultSet getColumns(JdbcTableHandle tableHandle, DatabaseMetaD
             return Optional.of(ColumnMapping.sliceMapping(
                     createUnboundedVarcharType(),
                     varcharReadFunction(),
-                    varcharWriteFunction(),
+                    (statement, index, value) -> {
+                        throw new PrestoException(NOT_SUPPORTED, "Underlying type is not supported for INSERT: " + typeHandle.getJdbcTypeName().orElse("?"));
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java]: `typeHandle.getJdbcTypeName().get()` + message saying that type is "forced mapped to varchar"

and `// TODO this should be handled during planning phase`

File: presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java [hunk 4]
```diff
@@ -320,6 +320,12 @@ public void testForcedMappingToVarchar()
                     sessionWithArrayAsArray(),
                     "SELECT * FROM tpch.test_forced_varchar_mapping",
                     "VALUES ('[\"2010-01-01 14:30:00\",\"2010-01-01 15:30:00\")','172.0.0.1',ARRAY['[\"2010-01-01 14:30:00\",\"2010-01-01 15:30:00\")'])");
+
+            // test insert into column that has forced varchar mapping
+            assertQueryFails(
+                    sessionWithArrayAsArray(),
```
Comment (findepi) [inline presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java]: use default session?

File: presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java [hunk 5]
```diff
@@ -321,6 +321,12 @@ public void testForcedMappingToVarchar()
                     "SELECT * FROM tpch.test_forced_varchar_mapping",
                     "VALUES ('[\"2010-01-01 14:30:00\",\"2010-01-01 15:30:00\")','172.0.0.1',ARRAY['[\"2010-01-01 14:30:00\",\"2010-01-01 15:30:00\")'])");
 
+            // test predicate pushdown to column that has forced varchar mapping
+            assertQuery(
+                    sessionWithArrayAsArray(),
```
Comment (findepi) [inline presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java]: default session

File: presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java [hunk 6]
```diff
@@ -321,6 +321,12 @@ public void testForcedMappingToVarchar()
...
+                    "SELECT * FROM tpch.test_forced_varchar_mapping WHERE tsrange_col = 'some value'",
+                    "SELECT 1 WHERE false");
```
Comment (findepi) [inline presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java]: add positive test case too

**Pull request merged**